pkg/snmp/snmpparse/config_snmp_test.go

```diff
@@ -108,6 +108,89 @@ func TestGetSNMPConfig(t *testing.T) {
 
 }
 
+func TestGetSNMPConfigNetwork(t *testing.T) {
+	IPList := []SNMPConfig{
+		{
+			Version:         "2",
+			CommunityString: "password",
+			NetAddress:      "192.168.5.0/24",
+			Port:            161,
+			Timeout:         60,
+			Retries:         3,
+		},
+		{
+			Version:         "2",
+			CommunityString: "drowssap",
+			IPAddress:       "98.6.18.159",
+			Port:            162,
+			Timeout:         30,
+			Retries:         5,
+		},
+		{
+			Version:         "3",
+			CommunityString: "drowssap",
+			IPAddress:       "98.6.18.160",
+			Port:            172,
+			Timeout:         30,
+			Retries:         5,
+		},
+	}
+	input := "192.168.5.3"
+	Exoutput := SNMPConfig{
+		Version:         "2",
+		CommunityString: "password",
+		IPAddress:       "192.168.5.3",
+		NetAddress:      "192.168.5.0/24",
+		Port:            161,
+		Timeout:         60,
+		Retries:         3,
+	}
+	assertIP(t, input, IPList, Exoutput)
+
+}
+
+func TestGetSNMPConfigNet(t *testing.T) {
+	//if the ip address is a part of network but alos is defined indivudualy
+	// the ip_address field should be the one that works
+	IPList := []SNMPConfig{
+		{
+			Version:         "2",
+			CommunityString: "password",
+			NetAddress:      "192.168.5.0/24",
+			Port:            161,
+			Timeout:         60,
+			Retries:         3,
+		},
+		{
+			Version:         "2",
+			CommunityString: "drowssap",
+			IPAddress:       "98.6.18.159",
+			Port:            162,
+			Timeout:         30,
+			Retries:         5,
+		},
+		{
+			Version:         "2",
+			CommunityString: "password",
+			IPAddress:       "192.168.5.1",
+			Port:            161,
+			Timeout:         60,
+			Retries:         3,
+		},
+	}
+	input := "192.168.5.1"
+	Exoutput := SNMPConfig{
+		Version:         "2",
+		CommunityString: "password",
+		IPAddress:       "192.168.5.1",
+		Port:            161,
+		Timeout:         60,
+		Retries:         3,
+	}
+	assertIP(t, input, IPList, Exoutput)
+
+}
+
```

Comment:
💬 I suggest adding a test for when both the `IPAddress` and the `NetAddress` are missing or for when the input doesn't match any of them (the case where you return an empty config).

Reply:
Will do 👌